### DISCUSSION TITLE
Make the cart rule more restrictive

### DISCUSSION THREAD

**Discussion by WiredSI** [PrestaShop/PrestaShop#34610] [Ideas]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

I created a free shipping cart rule that only applies if the products are from a certain supplier and if the total amount with VAT is more than €150. I noticed that if

add a new product to the same cart from a different supplier not selected in the restrictions to apply the rule, shipping remains free for all products.

![Senza titolo-1](https://github.com/PrestaShop/PrestaShop/assets/149384254/82ea66bf-043e-4237-ae7a-6227f0253303)

### Expected behavior

The rule should no longer apply if you add a product with a different supplier to your cart or apply the shipping cost only for the product(s) with a different supplier

### Steps to reproduce

1. Add two products to your cart with the supplier enabled in the free shipping cart rule.
2. Add a product with a different supplier that is not active in the cart rule restrictions.
3. Free shipping remains for all products including the product with different supplier.

### PrestaShop version(s) where the bug happened

8.1.1

### PHP version(s) where the bug happened

8.1

### Your company or customer's name goes here (if applicable).

Wired

**Comment by AureRita** [comment]
Hi @WiredSI ,

Thank you for your report, I reproduce the issue with PrestaShop version 8.1.2, as you can see : 

[Untitled_ Oct 31, 2023 4_04 PM.webm](https://github.com/PrestaShop/PrestaShop/assets/100699445/e3fc02dd-a6db-4448-a6c2-ba92c49fecc7)

Please be aware that some issues might take a very long time to be resolved.

@PrestaShop/product-council is this an expected behavior ? if not, I'll add this to the backlog so it can be fixed

Thanks

**Comment by Hlavtox** [comment]
It currently is an expected behavior, but I think cart rules can be greatly improved and this could be one of the scenarios.

Right now, if you define certain conditions, AS LONG AS THE CART MEETS THESE CONDITIONS, the cart rule will be applied. If there are additional products, it doesn't matter.

Expected behavior is to be specified here:
- Apply free shipping only to products that meet the criteria?
- Do not apply the cart rule at all?

**Comment by WiredSI** [comment]
> It currently is an expected behavior, but I think cart rules can be greatly improved and this could be one of the scenarios.
> 
> Right now, if you define certain conditions, AS LONG AS THE CART MEETS THESE CONDITIONS, the cart rule will be applied. If there are additional products, it doesn't matter.
> 
> Expected behavior is to be specified here:
> 
> * Apply free shipping only to products that meet the criteria?
> * Do not apply the cart rule at all?

The expected behavior is:

- Apply free shipping only to products that meet the criteria

**Comment by RosaBenouamer** [comment]
Hi @WiredSI 
I do agree that it seems quite logical "The rule should apply the shipping cost only for the product(s) with a different supplier" and has to be free only for the defined supllier but as @Hlavtox it has to be defined clearly in the documentation to share a common understanding :-)

**Comment by paulnoelcholot** [comment]
@MatShir, Does this sound good to you too?
